pkg/providers/pricing/controller.go

```diff
@@ -0,0 +1,63 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pricing
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+)
+
+type Controller struct {
+	pricingProvider *Provider
+}
+
+func NewController(pricingProvider *Provider) *Controller {
+	return &Controller{
+		pricingProvider: pricingProvider,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	c.updatePricing(ctx)
+
+	return reconcile.Result{RequeueAfter: 12 * time.Hour}, nil
+}
+
+func (c *Controller) Name() string {
+	return "pricing"
+}
+
+func (c *Controller) Builder(_ context.Context, m manager.Manager) corecontroller.Builder {
+	return corecontroller.NewSingletonManagedBy(m)
+}
+
+func (c *Controller) updatePricing(ctx context.Context) {
+	workqueue.ParallelizeUntil(ctx, 10, 1, func(i int) {
```

Comment:
`workqueue.ParallelizeUntil` should parallelize these two things that return errors. You can think of it like this

```go
work := []func(context.Context) error {
  c.pricingProvider.UpdateOnDemandPricing,
  c.pricingProvider.UpdateSpotPricing,
}
errs := make([]error, len(work))
workqueue.ParallizeUntil(ctx, len(work), len(work), func(i int) {
   errs[i] = work[i](ctx)
})
return multierr.Combine(errs...)
```

Reply:
Checkout Lo's parallel libraries. Easier to use imo